boost-gradle/openliberty-gradle/src/main/java/boost/runtimes/openliberty/LibertyGradleRuntime.java

```diff
@@ -0,0 +1,294 @@
+            }
+        } else {
+            throw new Exception(
+                    "No war files were found. The project must have a war packaging type or specify war dependencies.");
+        }
+
+        // Loop through configuration objects and add config and
+        // the corresponding Liberty feature
+        for (AbstractBoosterConfig configurator : boosterConfigurators) {
+            if (configurator instanceof LibertyBoosterI) {
+                ((LibertyBoosterI) configurator).addServerConfig(libertyConfig);
+                libertyConfig.addFeature(((LibertyBoosterI) configurator).getFeature());
+            }
+        }
+
+        libertyConfig.writeToServer();
+    }
+
+    //Liberty Gradle Plugin Execution
+
+    /**
+     * Invoke the liberty-gradle-plugin to run the install-feature goal.
+     *
+     * This will install any missing features defined in the server.xml or
+     * configDropins.
+     *
+     */
+    private void installMissingFeatures(Project project) throws GradleException {
+        runTask("installFeature", project);
+    }
+
+    /**
+     * Invoke the liberty-gradle-plugin to package the server into a runnable Liberty
+     * JAR
+     */
+    private void createUberJar(Project project) throws BoostException {
+        try {
+            runTask("libertyPackage", project);
+        } catch (GradleException e) {
+            throw new BoostException("Error creating Liberty uber jar", e);
+        }
+    }
+
+    public void doDebug(Object project, Object pluginTask) throws BoostException {
+        try {
+            runTask("libertyDebug", (Project)project);
+        } catch (GradleException e) {
+            throw new BoostException("Error debugging Liberty server", e);
+        }
+    }
+
+    public void doRun(Object project, Object pluginTask) throws BoostException {
+        try {
+            runTask("LibertyRun", (Project)project);
+        } catch (GradleException e) {
+            throw new BoostException("Error running Liberty server", e);
+        }
+    }
+
+    public void doStart(Object project, Object pluginTask) throws BoostException {
+        try {
+            runTask("libertyStart", (Project)project);
+        } catch (GradleException e) {
+            throw new BoostException("Error starting Liberty server", e);
+        }
+    }
+
+    public void doStop(Object project, Object pluginTask) throws BoostException {
+        try {
+            runTask("libertyStop", (Project)project);
+        } catch (GradleException e) {
+            throw new BoostException("Error stopping Liberty server", e);
+        }
+    }
+
+    private void runTask(String taskName, Project project) throws GradleException {
+        //String command = project.getProjectDir().toString() + "/gradlew";
+        try {
+            ProcessBuilder pb = new ProcessBuilder("gradle", taskName, "-i", "-s");
+            System.out.println("Executing task " + pb.command().get(1));
```

Comment:
Is this a pattern we've used anywhere before (kicking off another gradle invocation)?   Or have seen anywhere?  Just a bit hesitant to go too out on a limb.